### ISSUE TITLE
Crash on iOS 11 in [ABKIncrementEvent hash]

### ISSUE THREAD

**Issue by tastysauce** [Appboy/appboy-ios-sdk#120]
* [X] I have checked [Appboy's documentation](https://www.appboy.com/documentation/iOS/) and am still experiencing the issue.

# Report

## Describe your environment.

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | iOS
 Platform Version        | 11.0 Beta 5
 SDK Version             | 3.0.0
 Integration Method      | cocoapods
 Xcode Version           | Xcode 9 Beta 5
 Repro rate              | sometimes
 
## What did you do?

Launched the app

## What did you expect to happen?

App would get through `- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions`

## What happened instead?

App crashed

## Steps to reproduce

Unfortunately, we saw this crash reported 11 times for 1 user through Fabric's Crashlytics and not while the app was hooked up to Xcode.  The stack trace, however, points directly to your SDK and we're hoping that you find it useful.

## Code Snippet

Here is the stack trace:

```
Crashed: com.apple.main-thread
EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000000000001

0   libobjc.A.dylib  objc_msgSend + 8
1   Foundation  _NSDescriptionWithStringProxyFunc + 64
2   CoreFoundation __CFStringAppendFormatCore + 10140
3   CoreFoundation _CFStringCreateWithFormatAndArgumentsAux2 + 132
4   Foundation +[NSString stringWithFormat:] + 68
5   Appboy_iOS_SDK -[ABKIncrementEvent hash] + 140
6   CoreFoundation __21+[__NSSetI __new::::]_block_invoke + 52
7   CoreFoundation +[__NSSetI __new::::] + 368
8   CoreFoundation -[NSSet initWithArray:range:copyItems:] + 396
9   CoreFoundation +[NSSet setWithArray:range:copyItems:] + 72
10  CoreFoundation -[NSMutableArray removeObjectsInRange:inArray:range:] + 268
11  Appboy_iOS_SDK -[ABKEventLog removeEvents:] + 108
12  Appboy_iOS_SDK -[ABKPersistentData removeEventsFromEventLog:] + 156
13  Appboy_iOS_SDK -[ABKServerResponseHandler processResponseFromInFlightDataRequest:] + 708
14  Appboy_iOS_SDK -[ABKServerResponseHandler processResponse:withData:withInFlightRequest:] + 1548
15  Appboy_iOS_SDK __29-[ABKService processRequest:]_block_invoke.491 + 544
16  CFNetwork __75-[__NSURLSessionLocal taskForClass:request:uploadFile:bodyData:completion:]_block_invoke + 32
17  CFNetwork __49-[__NSCFLocalSessionTask _task_onqueue_didFinish]_block_invoke + 152
18  Foundation __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
19  Foundation -[NSBlockOperation main] + 72
20  Foundation -[__NSOperationInternal _start:] + 868
(Removed the rest)
```

## Are you doing any feature customizations that may relate to the issue? Can you share the code snippet?

No

## Project that demonstrates the issue

Our app is: https://itunes.apple.com/us/app/find-my-family-friends-phone-life360-locator/id384830320?mt=8
We can't provide a sample project that'll demonstrate the issue, unfortunately.

**Comment by Wenzhi** [comment]
Hi @tastysauce, 

Thanks for reporting the crash! We can reproduce it and are currently working on a fix for it. The fix will be released ASAP, and we will let you know when the fix is out.

Thanks,
Wenzhi

**Comment by tastysauce** [comment]
@Wenzhi That's great news!  Thanks a lot!

**Comment by Wenzhi** [comment]
Hey @tastysauce,

We just released SDK version 3.0.1. Could you check out the new SDK and see if it fixes the crashes?

Thanks,
Wenzhi

**Comment by Wenzhi** [comment]
Hey @tastysauce, 

I'm going to assume this is fixed for you given the inactivity. Please let us know if you see more issues or have any questions.

Thanks,
Wenzhi
